libraries/AP_Scheduler/AP_Scheduler.h

```diff
@@ -13,28 +13,19 @@
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-
-/*
- *  main loop scheduler for APM
- *  Author: Andrew Tridgell, January 2013
- *
- */
 #pragma once
 
-#include <AP_Param/AP_Param.h>
-#include <AP_HAL/Util.h>
+#include <stdint.h>
+#include <stdio.h>
 
-#define AP_SCHEDULER_NAME_INITIALIZER(_name) .name = #_name,
+#include <AP_HAL/Util.h>
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_Vehicle/AP_Vehicle.h>
 
-/*
-  useful macro for creating scheduler task table
- */
-#define SCHED_TASK_CLASS(classname, classptr, func, _rate_hz, _max_time_micros) { \
-    .function = FUNCTOR_BIND(classptr, &classname::func, void),\
-    AP_SCHEDULER_NAME_INITIALIZER(func)\
-    .rate_hz = _rate_hz,\
-    .max_time_micros = _max_time_micros\
-}
+// return the input parameter as a string
+#define xstr(s) str(s)
+#define str(s) #s
```

Comment:
macros in headers like this need to have namespace. It should actually be uppercase and probably be added to AP_Common.h, not here.